### PR TITLE
rvf for copyobj - CORTX-33463

### PR DESCRIPTION
**Pull request by archanabpatil**
Signed-off-by: archanabpatil <66412855+archanabpatil@users.noreply.github.com>

# Problem Statement
- RVF and TE for conditional parameters with copy object

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by archanabpatil** [comment]
Logs attached in jira : https://jts.seagate.com/browse/CORTX-33463

File: tests/s3/test_copy_object.py [hunk 1]
```diff
@@ -97,8 +96,7 @@ def setup(self):
         self.download_path = os.path.join(
             self.test_dir_path, self.object_name2)
         status, self.response1 = self.create_s3cortxcli_acc(
```
Comment (ketanarlulkar) [inline tests/s3/test_copy_object.py:98]: CLI Code should be removed

Comment (archanabpatil) [reply tests/s3/test_copy_object.py:98]: this creates iam accounts depending on the s3_engine types.

**Review by ketanarlulkar — APPROVED** [review]

File: tests/s3/test_copy_object.py [hunk 2]
```diff
@@ -2621,76 +2184,57 @@ def test_44756(self):
...
-        status, response = self.s3_obj.copy_object(self.bucket_name1, self.object_name1,
-                                                   self.bucket_name2, self.object_name2)
-        assert_utils.assert_true(status, response)
-        date1 = response["CopyObjectResult"]["LastModified"]
-        date1 = datetime.strptime(date1, '%b %d %Y')
-        pre_date = date1 - timedelta(days=1)
-        etag = response["CopyObjectResult"]["ETag"]
+        etag, pre_date, post_date = self.handle_copy(self.bucket_name1, self.object_name1,
+                                             self.bucket_name2, self.object_name2,
+                                             is_expect_err=False)
```
Comment (AshwiniS-Borse) [inline tests/s3/test_copy_object.py:2190]: indentation missing

File: tests/s3/test_copy_object.py [hunk 3]
```diff
@@ -2945,124 +2390,162 @@ def test_44767(self):
...
         LOGGER.info("ENDED: Test to validate the combinations of conditions with "
                     "x-amz-copy-source-if-none-match and x-amz-copy-source-if-unmodified-since for "
                     "copying simple and multipart object")
+
+    def upload_mpu_copy_obj(self, src_bucket, src_obj, dest_bucket, dest_obj, **kwargs):
+        """ method to upload object via multipart upload and copy it to some bucket
+        :param src_bucket: The name of the source bucket.
+        :param src_obj: The name of the source object.
+        :param dest_bucket: The name of the destination bucket.
+        :param dest_obj: The name of the destination object."""
```
Comment (AshwiniS-Borse) [inline tests/s3/test_copy_object.py:2461]: return param missing

File: tests/s3/test_copy_object.py [hunk 4]
```diff
@@ -2945,124 +2390,162 @@ def test_44767(self):
...
+                                                                 CopySourceIfModifiedSince=modified,
+                                                                 CopySourceIfMatch=match)
+                if unmodified is not None:
+                    status, response = self.s3_obj.copy_object(src_bucket, src_obj, dest_bucket,
+                                                               dest_obj, CopySourceIfMatch=match,
+                                                               CopySourceIfUnmodifiedSince=\
+                                                                   unmodified)
+                else:
+                    status, response = self.s3_obj.copy_object(src_bucket, src_obj, dest_bucket,
+                                                           dest_obj, CopySourceIfMatch=match)
```
Comment (AshwiniS-Borse) [inline tests/s3/test_copy_object.py:2504]: indentation missing.

File: tests/s3/test_copy_object.py [hunk 5]
```diff
@@ -2945,124 +2390,162 @@ def test_44767(self):
...
+                if modified is not None:
+                    status, c_response = self.s3_obj.copy_object(src_bucket, src_obj, dest_bucket,
+                                                                 dest_obj,
+                                                                 CopySourceIfModifiedSince=modified,
+                                                                 CopySourceIfNoneMatch=nonematch)
+                if unmodified is not None:
+                    status, c_response = self.s3_obj.copy_object(src_bucket, src_obj, dest_bucket,
+                                                                 dest_obj,
+                                                                 CopySourceIfUnmodifiedSince=\
+                                                                     unmodified,
```
Comment (AshwiniS-Borse) [inline tests/s3/test_copy_object.py:2515]: indentation missing.

File: tests/s3/test_copy_object.py [hunk 6]
```diff
@@ -2945,124 +2390,162 @@ def test_44767(self):
...
+                                                                 CopySourceIfNoneMatch=nonematch)
+                else:
+                    status, c_response = self.s3_obj.copy_object(src_bucket, src_obj, dest_bucket,
+                                                                 dest_obj,
+                                                                 CopySourceIfNoneMatch=nonematch)
+            elif modified is not None:
+                status, c_response = self.s3_obj.copy_object(src_bucket, src_obj, dest_bucket,
+                                                           dest_obj,
+                                                           CopySourceIfModifiedSince=modified)
```
Comment (AshwiniS-Borse) [inline tests/s3/test_copy_object.py:2524]: indentation missing.

File: tests/s3/test_copy_object.py [hunk 7]
```diff
@@ -2945,124 +2390,162 @@ def test_44767(self):
...
+            elif unmodified is not None:
+                status, c_response = self.s3_obj.copy_object(src_bucket, src_obj, dest_bucket,
+                                                           dest_obj,
+                                                           CopySourceIfUnmodifiedSince=unmodified)
```
Comment (AshwiniS-Borse) [inline tests/s3/test_copy_object.py:2528]: indentation missing.

File: tests/s3/test_copy_object.py [hunk 8]
```diff
@@ -2945,124 +2390,162 @@ def test_44767(self):
...
+            else:
+                response = self.s3_obj.create_bucket(self.bucket_name2)
+                assert_utils.assert_true(response[0], response[1])
+                status, c_response = self.s3_obj.copy_object(src_bucket, src_obj, dest_bucket,
+                                                           dest_obj)
```
Comment (AshwiniS-Borse) [inline tests/s3/test_copy_object.py:2533]: indentation missing.

**Review by AshwiniS-Borse — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**